src/votekit/pref_profile.py

```diff
@@ -95,29 +119,52 @@ def to_csv(self, fpath):
         Args:
             fpath (str): path to the saved csv
         """
-        with open(fpath, "w", newline="") as csvfile:
-            fieldnames = ["weight", "ranking"]
-            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
-            writer.writeheader()
-            for ballot in self.ballots:
-                writer.writerow({"weight": ballot.weight, "ranking": ballot.ranking})
+        if isinstance(self.ballots[0], Ballot):
```

Comment:
same as above comment, lots of duplication here when the two types can probably share most of the logic